### ISSUE TITLE
minikube with host docker override

### ISSUE THREAD

**Issue by sdhoward** [tilt-dev/tilt#4552]
I'm using minikube with docker runtime. In this configuration, Tilt opts to build using the docker daemon inside Minikube. This results in unbearably slow compile/cycle times for my Golang project (7 minutes on macos, 3 minutes on linux).

I want to [override the DOCKER_HOST variable](https://docs.tilt.dev/faq.html#q-how-do-i-tell-tilt-to-build-my-images-with-a-remote-docker-server). When I set it to `DOCKER_HOST=unix:///var/run/docker.sock`, I get more reasonable compile times (20-30 seconds) but the built **images will not deploy** to the cluster:

```
[K8s EVENT: Pod status-55b77854c7-plcg6 (ns: mynamespace)] Container image "192.168.49.2:5000/my_project:tilt-c823c5fb149773bf" is not present with pull policy of Never
```

This seems to be caused by:

> When you’re using Docker for Mac, Minikube, or MicroK8s, Tilt will automatically build the images in-cluster. When it detects this case, it will even modify your Kubernetes configs to set ImageNeverPull, so that Kubernetes will emit an error if it even tries to pull an image from a remote server.

[source](https://docs.tilt.dev/faq.html#q-all-the-tilt-examples-store-the-image-at-gcrio-isnt-it-really-slow-to-push-images-up-to-googles-remote-repository-for-local-development)

When I use k3d, I get the behavior that I want, where the images get built by the host daemon and uploaded to the registry inside.

For minikube, is there a way to toggle these two settings (ImagePullNever and DOCKER_HOST) individually, or am I doing something that will never work?

**Comment by nicks** [comment]
sorry to hear you're having trouble @sdhoward. Have you sen our cluster guide?
https://docs.tilt.dev/choosing_clusters.html

We generally don't recommend Minikube for new users - it has a ton of options that overlap in confusing ways, and it's easy to accidentally create a combination of options that's totally broken and/or has terrible performance.

But if you do want to try out Minikube, ctlptl can set it up for you correctly:
https://github.com/tilt-dev/ctlptl

This bash script is an abridged version of what ctlptl does:
https://github.com/tilt-dev/minikube-local
i.e., it turns off the docker daemon inside minikube and connects a registry to the cluster that you can push to

**Comment by sdhoward** [comment]
I was originally using ctlptl and wanted to configure a registry mirror. I thought I needed to supply the minikube option for `--registry-mirror`. Therefore I moved off of ctlptl and was using the minikube command directly.

But I was wrong, ctlptl makes minikube use the host docker daemon for building like you said. So that solves both of my problems: my build times and my registry mirror setting. Thanks for the explanation of what ctlptl does.
